src/special/karma.js

```diff
@@ -0,0 +1,160 @@
+import path from 'path';
+import { evaluate, tryRequire } from '../utils';
+
+// TODO support karma.conf.coffee
+const supportedConfNames = [
+  'karma.conf.js',
+  'karma.conf.ts',
+  '.config/karma.conf.js',
+  '.config/karma.conf.ts',
+];
+
+function parseConfig(content) {
+  try {
+    const confSetter = evaluate(content);
+    let conf = {};
+    confSetter({
+      set(c) {
+        conf = c;
+      },
+    });
+    return conf;
+  } catch (error) {
+    // not valid JavaScript code
+  }
+
+  // parse fail, return nothing
+  return {};
+}
+
+function wrapToArray(obj) {
+  if (!obj) {
+    return [];
+  }
+  if (Array.isArray(obj)) {
+    return obj;
+  }
+
+  return [obj];
+}
+
+function wrapToMap(obj) {
+  if (!obj) {
+    return {};
+  }
+  return obj;
+}
+
+function collectFrameworks(config, karmaPluginsInstalled) {
+  // karma-x plugins define frameworks using the structure { 'framework:<name>': ['factory', f] }
+  // generate a lookup map { '<name>': 'karma-x', ... }
+  const frameworkMapping = {};
+  // skip plugins with more specific names, such as launchers and reporters
+  karmaPluginsInstalled.filter(plugin => !plugin.endsWith('-launcher') && !plugin.endsWith('-reporter')).forEach((genericPlugin) => {
+    const p = tryRequire(genericPlugin);
```

Comment:
This specific approach - using node's require to try to pull in the relevant module - seems to be against the philosophy of the rest of the project. Sadly the information we need to resolve frameworks/browsers/reporters/etc to plugins isn't quite available statically, which means we need to find the default module in each plugin and parse out the map from its module.export. Any guidance on how to do that cheaply would be much appreciated!

Reply:
Opted to replace it with parser code, so I can guarantee it works in my own project.